blogposts/2021/release-3.31.md

```diff
@@ -0,0 +1,212 @@
+    category: 
+  - description: "Sourcegraph's Alertmanager dependency has been upgraded to v0.22.2. 23663"
+    url: https://github.com/sourcegraph/sourcegraph/pull/23714
+    category: 
+  - description: Code Insights will now schedule sample recordings for the first of the next month after creation or a previous recording.
+    url: https://github.com/sourcegraph/sourcegraph/pull/23799
+    category: 
+  - description: Code Insights now stores data in a new format. Data points will store complete vectors for all repositories even if the underlying Sourcegraph queries were compressed.
+    url: https://github.com/sourcegraph/sourcegraph/pull/23768
+    category: Repositories
+  - description: Code Insights rate limit values have been tuned for a more reasonable performance.
+    url: https://github.com/sourcegraph/sourcegraph/pull/23860
+    category: 
+  - description: "Code Insights will now generate historical data once per month on the first of the month, up to the configured `insights.historical.frames` number of frames."
+    url: https://github.com/sourcegraph/sourcegraph/pull/23768
+    category: 
+  - description: Code Insights will now schedule recordings for the first of the next calendar month after an insight is created or recorded.
+    url: https://github.com/sourcegraph/sourcegraph/pull/23799
+    category: 
+  - description: Code Insights will attempt to sync insight definitions from settings to the database once every 10 minutes. 23805
+    url: https://github.com/sourcegraph/sourcegraph/pull/23805
+    category: 
+  - description: "Code Insights exposes information about queries that are flagged `dirty` through the `insights` GraphQL query."
+    url: https://github.com/sourcegraph/sourcegraph/pull/23857/
+    category: API
+  - description: "Code Insights GraphQL query `insights` will now fetch 12 months of data instead of 6 if a specific time range is not provided."
+    url: https://github.com/sourcegraph/sourcegraph/pull/23786
+    category: API
+  - description: Code Insights will now generate 12 months of historical data during a backfill instead of 6.
+    url: https://github.com/sourcegraph/sourcegraph/pull/23860
+    category: 
+  - description: "The `sourcegraph-frontend.Role` in Kubernetes deployments was updated to permit statefulsets access in the Kubernetes API. This is needed to better support stable service discovery for stateful sets during deployments, which isn't currently possible by using service endpoints. [#3670](https://github.com/sourcegraph/deploy-sourcegraph/pull/3670)"
+    url: https://github.com/sourcegraph/sourcegraph/pull/23889
+    category: 
+  - description: "For Docker-Compose and Kubernetes users, the built-in main Postgres and codeintel databases have switched to an alpine Docker image. This requires re-indexing the entire database. This process can take up to a few hours on systems with large datasets."
+    url: https://github.com/sourcegraph/sourcegraph/pull/23697
+    category: 
```

Comment:
```suggestion
  - description: "For Docker-Compose and Kubernetes users, the built-in main Postgres and codeintel databases have switched to an alpine Docker image. This requires re-indexing the entire database. This process can take up to a few hours on systems with large datasets. [Learn more](https://docs.sourcegraph.com/admin/migration/3_31)."
    url: https://github.com/sourcegraph/sourcegraph/pull/23697
    category: Admin
```

Reply:
cc @ggilmore might want to add a callout for this migration requirement in this blog post!